crates/fuel-core/src/graphql_api/storage.rs

```diff
@@ -249,4 +249,53 @@ where
         self.commit()?;
         Ok(())
     }
+
+    #[cfg(feature = "fault-proving")]
+    fn registry_root(&self) -> StorageResult<Bytes32> {
+        use fuel_core_storage::tables::merkle::DenseMetadataKey;
+
+        use crate::graphql_api::storage::da_compression::v2::{
+            address::TemporalRegistryAddressMerkleMetadata as AddressMerkle,
+            asset_id::TemporalRegistryAssetIdMerkleMetadata as AssetIdMerkle,
+            contract_id::TemporalRegistryContractIdMerkleMetadata as ContractIdMerkle,
+            evictor_cache::TemporalRegistryEvictorCacheMerkleMetadata as EvictorCacheMerkle,
+            predicate_code::TemporalRegistryPredicateCodeMerkleMetadata as PredicateCodeMerkle,
+            registry_index::TemporalRegistryIndexMerkleMetadata as RegistryIndexMerkle,
+            script_code::TemporalRegistryScriptCodeMerkleMetadata as ScriptCodeMerkle,
+            timestamps::TemporalRegistryTimestampsMerkleMetadata as TimestampsMerkle,
+        };
+
+        macro_rules! get_merkle_root {
+            ($self:ident, $metadata:ty) => {
+                $self
+                    .storage::<$metadata>()
```

Comment:
Why don't we call the provided `.root()` method from the `SupportsMerkle` trait here? The implementation seems basically the same except we hard-code `DenseMetadataKey::Latest` which probably is fine but has a bit of code smell to it.

Reply:
this way i don't have to keep track of the latest key, as you can see in -
https://github.com/FuelLabs/fuel-core/blob/0dd6795ebfc205b0b84f5ffc47d027282a532528/crates/storage/src/blueprint/merklized.rs#L233-L241

it reuses the key as the `PrimaryKey` variant of the `DenseMetadataKey` enum

Reply:
Yeah, I just think it's nice to explicitly provide the key but this is probably fine - and we can change it later if we need it.